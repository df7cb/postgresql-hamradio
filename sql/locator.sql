SELECT 'JO'::locator;
SELECT 'JO31'::locator;
SELECT 'JO31hi'::locator;
SELECT 'JO31HI'::locator;
SELECT 'JO31hi19'::locator;
SELECT 'JO31hi19lj'::locator;
SELECT ST_Locator('JO');
SELECT ST_Locator('JO31');
SELECT ST_Locator('JO31hi');
SELECT ST_Locator('JO31hi19');
SELECT ST_Locator('JO31hi19lj');
SELECT ST_LocatorPoint('JO');
SELECT ST_LocatorPoint('JO31');
SELECT ST_LocatorPoint('JO31hi');
SELECT ST_LocatorPoint('JO31hi19');
SELECT ST_LocatorPoint('JO31hi19lj');
SELECT field FROM locator2 WHERE ST_Intersects(geom, ST_SetSRID(ST_Point(6.629, 51.356), 4326));
SELECT field FROM locator4 WHERE ST_Intersects(geom, ST_SetSRID(ST_Point(6.629, 51.356), 4326));

-- errors
SELECT 'J'::locator;
SELECT 'JO3'::locator;
SELECT 'jo31'::locator;
