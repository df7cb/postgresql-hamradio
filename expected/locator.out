SELECT 'JO'::locator;
 locator 
---------
 JO
(1 row)

SELECT 'JO31'::locator;
 locator 
---------
 JO31
(1 row)

SELECT 'JO31hi'::locator;
 locator 
---------
 JO31hi
(1 row)

SELECT 'JO31HI'::locator;
 locator 
---------
 JO31HI
(1 row)

SELECT 'JO31hi19'::locator;
 locator  
----------
 JO31hi19
(1 row)

SELECT 'JO31hi19lj'::locator;
  locator   
------------
 JO31hi19lj
(1 row)

SELECT ST_Locator('JO');
                                                                                             st_locator                                                                                             
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 0103000020E61000000100000005000000000000000000000000000000000049400000000000003440000000000000494000000000000034400000000000004E4000000000000000000000000000004E4000000000000000000000000000004940
(1 row)

SELECT ST_Locator('JO31');
                                                                                             st_locator                                                                                             
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 0103000020E61000000100000005000000000000000000184000000000008049400000000000002040000000000080494000000000000020400000000000004A4000000000000018400000000000004A4000000000000018400000000000804940
(1 row)

SELECT ST_Locator('JO31hi');
                                                                                             st_locator                                                                                             
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 0103000020E610000001000000050000005255555555551A40A6AAAAAAAAAA4940AEAAAAAAAAAA1A40A6AAAAAAAAAA4940AEAAAAAAAAAA1A400000000000B049405255555555551A400000000000B049405255555555551A40A6AAAAAAAAAA4940
(1 row)

SELECT ST_Locator('JO31hi19');
                                                                                             st_locator                                                                                             
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 0103000020E61000000100000005000000E2DDDDDDDD5D1A407377777777AF49406666666666661A407377777777AF49406666666666661A400000000000B04940E2DDDDDDDD5D1A400000000000B04940E2DDDDDDDD5D1A407377777777AF4940
(1 row)

SELECT ST_Locator('JO31hi19lj');
                                                                                             st_locator                                                                                             
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 0103000020E610000001000000050000001BC7711CC7611A40A6AAAAAAAAAF49401E22222222621A40A6AAAAAAAAAF49401E22222222621A4055B0055BB0AF49401BC7711CC7611A4055B0055BB0AF49401BC7711CC7611A40A6AAAAAAAAAF4940
(1 row)

SELECT ST_LocatorPoint('JO');
                  st_locatorpoint                   
----------------------------------------------------
 0101000020E610000000000000000024400000000000804B40
(1 row)

SELECT ST_LocatorPoint('JO31');
                  st_locatorpoint                   
----------------------------------------------------
 0101000020E61000000000000000001C400000000000C04940
(1 row)

SELECT ST_LocatorPoint('JO31hi');
                  st_locatorpoint                   
----------------------------------------------------
 0101000020E61000000000000000801A405A55555555AD4940
(1 row)

SELECT ST_LocatorPoint('JO31hi19');
                  st_locatorpoint                   
----------------------------------------------------
 0101000020E61000001E22222222621A40C0BBBBBBBBAF4940
(1 row)

SELECT ST_LocatorPoint('JO31hi19lj');
                  st_locatorpoint                   
----------------------------------------------------
 0101000020E61000009DF4499FF4611A40842DD882ADAF4940
(1 row)

SELECT field FROM locator2 WHERE ST_Intersects(geom, ST_SetSRID(ST_Point(6.629, 51.356), 4326));
 field 
-------
 JO
(1 row)

SELECT field FROM locator4 WHERE ST_Intersects(geom, ST_SetSRID(ST_Point(6.629, 51.356), 4326));
 field 
-------
 JO31
(1 row)

-- errors
SELECT 'J'::locator;
ERROR:  value for domain locator violates check constraint "valid_locator"
SELECT 'JO3'::locator;
ERROR:  value for domain locator violates check constraint "valid_locator"
SELECT 'jo31'::locator;
ERROR:  value for domain locator violates check constraint "valid_locator"
